src/cloud-api-adaptor/test/provisioner/kustomize.go

```diff
@@ -170,6 +170,57 @@ func (kh *KustomizeOverlay) SetKustomizeSecretGeneratorLiteral(secretName string
 	return nil
 }
 
+// SetKustomizeSecretGeneratorEnvs updates the kustomization YAML by adding the `env` on the
+// `sgName` SecretGenerator env file.
+func (kh *KustomizeOverlay) SetKustomizeSecretGeneratorEnv(sgName string, file string) (err error) {
```

Comment:
Is this a good opportunity for some refactoring as now `SetKustomizeSecretGeneratorEnv`, `SetKustomizeSecretGeneratorLiteral` and `SetKustomizeSecretGeneratorFile` have pretty similar code duplicated?

Reply:
Hi @stevenhorsman ! I looked at kustomize.go and it really have a bunch of duplicated code. I plan to refactor it but on a separated PR because I will need to re-run the tests. Is it okay for you?

Reply:
Yeah, that's fine